stock_request/models/stock_request_order.py

```diff
@@ -50,7 +51,7 @@ def _get_default_requested_by(self):
         states={'draft': [('readonly', False)]})
     location_id = fields.Many2one(
         'stock.location', 'Location', readonly=True,
-        domain=[('usage', 'in', ['internal', 'transit'])],
+        domain=[('usage', 'in', ['internal', 'transit', 'customer'])],
```

Comment:
@etobella Because we use stock requests in field service and field service locations sometimes belong to the customer.

Reply:
But you cannot change the behavior of a generic module for an specific module. If you need this change, make it there.

Reply:
@max3903 IMHO this would cause mistakes from users that would incorrectly enter a customer location, leading to undesired results. 

I am sure that this makes complete sense to your use case, but perhaps the field 'direction' or the 'customer' location type  should be added in a field service specific module, or even should be part of your own customization.

Reply:
As I said in other comments, If you need this kind of changes, you should make in specific modules. You are creating a hole for most companies because field service companies uses this behavior. I think it is a bad idea for a generic module as `stock_request`. What do you think @jbeficent as creator of this module?

Reply:
@etobella I am sorry. I am changing the behavior of a specific module (that only supports internal moves) to be more generic (and supports external ones as well).

Reply:
@max3903 You should be able to see that a module that allows users to make external account moves without using sales or invoices can be a problem for a lot of companies. I think this a good reason to notice the problem and think if we need a new module.

I know that sometimes comments are not satisfying for everyone, but we should try to think for all the people that uses OCA, not only our customers or company.

Reply:
@max3903 we implemented a setting that allows users to enter any location, not only internal. You have to activate it here:
https://github.com/OCA/stock-logistics-warehouse/blob/11.0/stock_request/models/res_config_settings.py#L19

And it's used here:
https://github.com/OCA/stock-logistics-warehouse/blob/11.0/stock_request/models/stock_request_abstract.py#L188

So, a normal user would not see this 'special' behavior unless they activate the setting.

You could make use of this feature, or we can find a way that works for all.